File: gazda/nekretnine/report/brojila/brojila.py
# Copyright (c) 2025, Filip Ilic and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
	columns = [
		{
			"fieldname": "naziv_nekretnine",
			"label": "Naziv nekretnine",
			"fieldtype": "Link",
			"options": "Nekretnina",
			"width": 350
		},
		{
			"fieldname": "zakupac",
			"label": "Zakupac",
			"fieldtype": "Link",
			"options": "Osoba",
			"width": 180
		},
		{
			"fieldname": "tip_brojila",
			"label": "Tip brojila",
			"fieldtype": "Link",
			"options": "Tip Transakcije",
			"width": 150
		},
		{
			"fieldname": "broj_brojila",
			"label": "Broj brojila",
			"fieldtype": "Data",
			"width": 350
		}
	]

	data = frappe.db.sql("""
		SELECT 
			n.name as naziv_nekretnine,
			n.zakupac,
			b.tip_brojila,
			b.broj_brojila
		FROM 
			`tabNekretnina` n
		LEFT JOIN 
			`tabNekretnina Brojila` b ON b.parent = n.name
		WHERE 
			b.tip_brojila IS NOT NULL
		ORDER BY 
			n.naziv_nekretnine, b.tip_brojila
	""", as_dict=1)

	return columns, data

File: gazda/nekretnine/report/brojila/brojila.js
// Copyright (c) 2025, Filip Ilic and contributors
// For license information, please see license.txt

frappe.query_reports["Brojila"] = {
	"filters": [],
	"formatter": function(value, row, column, data, default_formatter) {
		if (column.fieldname === "tip_brojila") {
			return `<div class="pill">${value}</div>`;
		}
		return default_formatter(value, row, column, data);
	},
	onload: function(report) {
		// Add CSS for pills
		$('head').append(`
			<style>
				.pill {
					display: inline-block;
					background-color: var(--orange-100);
					color: var(--orange-600);
					padding: 2px 8px;
					border-radius: 12px;
					font-size: 14px;
					white-space: nowrap;
				}
			</style>
		`);
	}
};
